quint/src/runtime/trace.ts

```diff
@@ -331,10 +327,17 @@ class TraceRecorderImpl implements TraceRecorder {
     traceToSave.result = outcome
     traceToSave.args = trace
 
-    this.traces.push({ frame: traceToSave, seed: this.runSeed })
+    const traceWithSeed = { frame: traceToSave, seed: this.runSeed }
+
+    this.bestTraces.push(traceWithSeed)
+    this.sortTracesByQuality()
+    // Remove the worst trace (if there more traces than needed)
+    if (this.bestTraces.length > this.tracesToStore) {
+      this.bestTraces.pop()
+    }
```

Comment:
This PR has been merged already. However, here is my concern. Now best traces are sorted on every run, even though only one new trace is added on every run. I don't know how large `this.tracesToStore` would be, but assuming it is `n`, we would have something like `O(n * log(n))` operations spent on sorting in every run. Instead, you could just do one step of insertion sort, assuming that `this.bestTraces` is sorted already. This would give us a simple loop of `n/2` iterations on average.

Reply:
Yes, I'm aware. But as I tried explaining in the description, `n=1` in all existing use cases. It is only greater than 1 if you use the new CLI param `--n-traces`. 

I did look up insertion sort implementations in typescript to use here, but there was nothing very handy and I just wanted to fix the memory issue. This seems like a fun thing to improve and benchmark tho, and my students will need to use this `--n-traces` features by the end of the semester, so I'll take care of performance eventually :grimacing:

Reply:
I mean, you don't need any special implementation for insertion sorting. To insert a new element `e`, you just do one iteration over the array `x` and find the index `i` such that `x[i] <= e <= x[i+1]` (of course, you have to take care of the cases `i == 0` and `i == x.length`). Then, you insert `e` at the position `i`. I guess, JS `splice` should be able to do that.